src/item/LingeringPotion.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\entity\Location;
+use pocketmine\entity\projectile\SplashPotion as SplashPotionEntity;
+use pocketmine\entity\projectile\Throwable;
+use pocketmine\player\Player;
+
+class LingeringPotion extends ProjectileItem{
+
+	private PotionType $potionType = PotionType::WATER;
+
+	protected function describeState(RuntimeDataDescriber $w) : void{
+		$w->enum($this->potionType);
+	}
+
+	public function getType() : PotionType{ return $this->potionType; }
+
+	/**
+	 * @return $this
+	 */
+	public function setType(PotionType $type) : self{
+		$this->potionType = $type;
+		return $this;
+	}
+
+	public function getMaxStackSize() : int{
+		return 1;
+	}
+
+	protected function createEntity(Location $location, Player $thrower) : Throwable{
+		$projectile = new SplashPotionEntity($location, $thrower, $this->potionType);
+		$projectile->setLinger();
```

Comment:
I wonder if we could just add a `bool $linger` parameter to the constructor of `SplashPotion` for this. The two classes are identical apart from this function.

Reply:
Adding it on constructor since the class is not final would be a BC break for inheritors. but we could try a `setLinger()` method.

Reply:
It's not BC breaking unless it's required.

Reply:
`setLinger()` isn't an option because we already have a type ID constant for lingering potion. It would be a BC break to remove it, and confusing to plugin devs not to remove it.